[QA] 연락처 외 수정사항 반영

rhkrwngud445 opened the pull request:
## 작업 내용
#58 에 해당하는 기획/디자인 QA 수정하였습니다!

## 확인 방법
Preview 및 구동

## 참고 사항
가능하시다면 디바이스 더블 체크 부탁드립니다!
그리고 Friend 재활용을 해보려하는데, God Object가 되어가네요.. 추후 UI 모델로 분리해보겠습니다!

## 관련 이슈
- close #58

stopstone committed:
feat: 이번달 챙길사람 전체보기 화면 네비게이션 추가

- 홈 화면의 '이달의 리마인더' 섹션에 '전체보기' 버튼을 추가하고, 클릭 시 해당 화면으로 이동하도록 구현했습니다.
- `ic_front_8` 벡터 드로어블을 추가하여 '전체보기' 버튼 아이콘으로 사용했습니다.
- 전체보기 화면으로 이동하는 `navigateToMonthlyReminderAll` 네비게이션 로직과 `MonthlyReminderAllScreen` 컴포저블을 추가했습니다.

stopstone committed:
feat: 월별 리마인더 친구 카드 컴포저블 추가

- 월별 전체 리마인더 화면에서 사용될 `MonthlyReminderFriendCard` 컴포저블을 추가했습니다.
- 이 카드는 친구의 이름, 리마인더 내용, D-day, 그리고 '챙김 기록하기' 버튼으로 구성됩니다.

stopstone committed:
feat: 월별 리마인더 화면 빈 상태 컴포저블 추가

- 월별 리마인더에 표시할 내용이 없을 때 보여줄 `MonthlyReminderEmpty` 컴포저블을 추가했습니다.
- 해당 컴포저블은 안내 이미지와 "이번달은 챙길 사람이 없네요."라는 텍스트로 구성됩니다.

stopstone committed:
feat: 월별 리마인더 완료 목록 아이템 컴포저블 추가

- 월별 리마인더의 완료된 항목을 표시하는 `MonthlyReminderComplete` 컴포저블을 새로 추가했습니다.
- 이 컴포저블은 아이콘, 이름, 날짜를 포함하며, 완료된 상태의 UI를 나타냅니다.

stopstone committed:
refactor: 월별 리마인더 전체보기 화면 파일 이동 및 ViewModel 연결

stopstone committed:
feat: 이번달 챙길 사람 전체보기 화면 구현

- UI 상태(로딩, 데이터 있음, 데이터 없음)에 따라 각각 다른 화면을 표시합니다.
- 뒤로가기 버튼을 통해 이전 화면으로 이동할 수 있습니다.

stopstone committed:
feat: 이달의 챙김 친구 카드 UI 모델 적용

stopstone committed:
feat: iewModel 및 UI State 구현

- `MonthlyReminderAllViewModel`을 추가하여 월별 리마인더 목록을 가져오고 UI 상태를 관리합니다.
- API 응답을 `MonthlyReminderUIModel`로 변환하여 UI에 필요한 데이터를 제공합니다.
- `MonthlyReminderAllUIState` sealed interface를 `Loading`, `Empty`, `Success` 상태로 정의하여 UI 상태를 관리하도록 수정했습니다.

stopstone committed:
feat: 이달의 챙김 완료 친구 목록 조회 API 추가

stopstone committed:
chore: MonthlyReminderFriendCard 패딩 삭제

stopstone committed:
feat: 챙김 완료 목록 아이템 UI 구현

- `MonthlyReminderComplete` 컴포저블이 `MonthlyReminderUIModel` 데이터를 받아 실제 데이터를 표시하도록 수정했습니다.
- 날짜 형식을 'yy.MM.dd'로 포맷하는 `formatDate` 함수를 추가했습니다.
- 이름이 길어질 경우 말줄임표(...)로 처리되도록 `TextOverflow.Ellipsis` 속성을 적용했습니다.

stopstone committed:
feat: 챙김 완료 목록 추가

stopstone committed:
feat: 연락 기록 성공 다이얼로그 추가

- 연락을 기록했을 때 표시되는 성공 다이얼로그 UI를 구현했습니다.
- 다이얼로그는 1.5초 후 자동으로 사라집니다.

stopstone committed:
refactor: 챙김 친구 완료 시 UI 즉시 업데이트

- 월별 리마인더 목록에서 완료 처리한 친구를 연락 완료 목록으로 즉시 이동시켜 UI에 반영하도록 수정했습니다.
- 기존의 네트워크 통신을 통한 목록 갱신 방식 대신, 로컬에서 목록을 직접 업데이트하여 불필요한 API 호출을 제거하고 응답성을 개선했습니다.

stopstone committed:
refactor: 월별 리마인더 없음 화면 UI 수정

stopstone committed:
refactor: 이번달 챙길 사람 화면 문자열 리소스 적용

stopstone committed:
refactor: 이번달 챙길 사람 화면 챙김 완료 상단 패딩 수정

- 불필요한 Spacer를 제거하여 레이아웃을 개선했습니다.
- LazyColumn의 컨텐츠 패딩을 적용하여 상하단 여백을 조정했습니다.

stopstone committed:
refactor: 연락 완료 여부 확인 로직을 uiState로 관리

stopstone committed:
refactor: 연락처에서 불러오기 그림자 효과 수치값 수정

stopstone committed:
refactor: CycleSettingBottomSheet 주기설정 공백 추가

stopstone committed:
feat: 주기 설정 바텀 시트 텍스트 동적 변경

- 기존에 '매주'로 고정되어 있던 주기 설정 바텀 시트의 텍스트를 `selectedInterval` 값에 따라 '매주', '2주마다', '매달'로 동적으로 변경되도록 수정했습니다.
- `DateExtension.getCycleText()`를 사용하여 선택된 주기에 맞는 텍스트를 가져오도록 구현했습니다.
- 주기 텍스트에서 요일 부분에만 다른 스타일이 적용되도록 로직을 개선했습니다.

stopstone committed:
refactor: 연락 주기 설정 화면 '다음 주기' 텍스트 strings.xml 관리

stopstone committed:
refactor: 연락 주기 화면 UI 패딩값 수정

stopstone committed:
refactor: 검색 아이콘 색상 변경

stopstone committed:
refactor: 연락처 불러오기 기본 이미지 변경

stopstone committed:
refactor: 이번달 챙길 사람 화면 간격 추가

- 월별 리마인더 전체보기 화면의 카드 사이에 16dp의 간격을 추가했습니다.

stopstone committed:
feat: 이번달 챙길 사람 리스트 정렬

- 월별 리마인더 목록을 `nextContactAt`(다음 연락일) 기준으로 오름차순 정렬하여 표시하도록 수정했습니다.

rhkrwngud445 committed:
feat: shimmer modifier 확장자 추가

rhkrwngud445 committed:
feat: shimmer modifier 확장자 추가

rhkrwngud445 committed:
fix: 가로 shimmer로 수정

rhkrwngud445 committed:
fix: 가로 shimmer로 수정

rhkrwngud445 committed:
refactor: shimmer 적용 및 home uiState로 상태 단일 관리

rhkrwngud445 committed:
refactor: shimmer 적용 및 home uiState로 상태 단일 관리

rhkrwngud445 committed:
fix: 이번 달 챙길 사람 우측 패딩 삭제

rhkrwngud445 committed:
fix: 이번 달 챙길 사람 우측 패딩 삭제

rhkrwngud445 committed:
chore: home preview 데이터 수정

rhkrwngud445 committed:
chore: home preview 데이터 수정

stopstone committed:
refactor: 월별 리마인더 데이터 로딩 로직 개선

- `combine`을 사용하여 `fetchMonthlyFriends`와 `fetchMonthlyCompleteFriends` API 호출을 병렬로 처리하도록 수정했습니다.
- 두 API의 응답을 모두 받은 후에 한 번에 UI 상태를 업데이트하여 데이터 로딩 중 발생할 수 있는 화면 깜빡임 현상을 개선했습니다.
- API 응답을 함께 관리하기 위한 `MonthlyReminderCombinedData` data class를 추가했습니다.

stopstone committed:
refactor: 홈 화면 "전체 보기" 텍스트를 문자열 리소스로 분리

stopstone committed:
refactor: 에러 처리 방식 개선

stopstone committed:
refactor: Flow 에러 처리 로직 공통화

- 반복적으로 사용되는 Flow의 `catch` 블록 내 에러 처리 로직을 `handleError` 확장 함수로 분리하여 공통화했습니다.
- `MonthlyReminderAllViewModel` 내 `fetchMonthlyReminders`와 `onRecordFriendShip` 함수의 에러 처리 로직에 `handleError`를 적용하여 코드 중복을 제거하고 가독성을 개선했습니다.

stopstone committed:
refactor: 기록 완료 다이얼로그 색상 수정

- `RecordSuccessDialog`에서 사용하던 하드코딩된 색상 값을 `NearTheme`의 색상으로 변경했습니다.
- `NearColor`에 `BLACK_222222` 색상을 추가했습니다.

stopstone committed:
chore: MonthlyReminderFriendCard.kt 코드 스타일 수정

stopstone committed:
refactor: 이번달 챙길사람 하단 패딩 수정

stopstone committed:
chore: "D-day"를 "D-DAY"로 수정

- 연락 주기까지 남은 날짜를 표시하는 문자열을 "D-day"에서 "D-DAY"로 변경했습니다.

stopstone committed:
feat: 연락일이 오늘일 경우 D-day 스타일 적용 및 중복 데이터 제거

- `MonthlyReminderUIModel`에 연락일이 오늘인지 여부를 나타내는 `isToday` 프로퍼티를 추가했습니다.
- `isToday` 값에 따라 D-day 텍스트 스타일(파란색, 굵은 글씨)이 적용되도록 `MonthlyReminderFriendCard`를 수정했습니다.
- 월간 리마인더 목록에서 중복된 친구 데이터가 표시되지 않도록 `distinctBy`를 사용하여 필터링 로직을 추가했습니다.

stopstone committed:
refactor: MonthlyReminderAllScreen의 불필요한 launch 제거

`LaunchedEffect` 내에서 `collect`를 사용할 때 불필요한 `launch` 블록을 제거하여 코드를 간소화했습니다.

stopstone committed:
refactor: MonthlyReminderAllViewModel 내 불필요한 private backing property 제거

- `_monthlyReminders`와 `_completedReminders`에 대한 불필요한 `_` 접두사를 제거하여 코드를 간소화했습니다.

stopstone committed:
refactor: updateUIState 함수의 이름 변경

- `updateUIState` 함수의 역할을 더 명확하게 나타내기 위해 `combineRemindersToUIState`로 이름을 변경했습니다.

rhkrwngud445 committed:
fix: inset 패딩 적용 및 작은 높이 화면 대응

rhkrwngud445 committed:
fix: inset 패딩 적용 및 작은 높이 화면 대응

rhkrwngud445 committed:
fix: login 이미지, 텍스트 이미지 간격 수정

rhkrwngud445 committed:
fix: login 이미지, 텍스트 이미지 간격 수정

rhkrwngud445 committed:
fix: 연락처 검색 버튼 색상 수정

rhkrwngud445 committed:
fix: 연락처 검색 버튼 색상 수정

rhkrwngud445 committed:
fix: home 디바이스 긴 경우 중간 마진 추가

- 타이틀 ~ 이번달 챙길사람 패딩 작업

rhkrwngud445 committed:
fix: My 로그아웃 버튼 패딩 수정

rhkrwngud445 committed:
fix: My 로그아웃 버튼 패딩 수정

rhkrwngud445 committed:
fix(NearOutlinedTextField): 에러시 container 색상이 변하지 않도록 수정

rhkrwngud445 committed:
fix(NearOutlinedTextField): 에러시 container 색상이 변하지 않도록 수정

rhkrwngud445 committed:
fix(Friend): DateTime 계산 로직을 Data 레이어에서 수행하도록 수정

- preview에서 DateTime 계산으로 렌더링이 안되는 문제 개선

rhkrwngud445 committed:
fix(Friend): DateTime 계산 로직을 Data 레이어에서 수행하도록 수정

- preview에서 DateTime 계산으로 렌더링이 안되는 문제 개선

rhkrwngud445 committed:
fix: FriendProfile 스크롤 범위 수정

rhkrwngud445 committed:
fix: FriendProfile 스크롤 범위 수정

rhkrwngud445 committed:
feat: 챙김 기록하기 버튼 그라데이션 배경 적용

rhkrwngud445 committed:
feat: 챙김 기록하기 버튼 그라데이션 배경 적용

rhkrwngud445 committed:
fix: 친구 프로필 수정 메모와 BasicText간 간격 수정

rhkrwngud445 committed:
fix: 친구 프로필 수정 메모와 BasicText간 간격 수정

rhkrwngud445 committed:
fix: UI 표시되는 날짜 포맷 수정

rhkrwngud445 committed:
fix: UI 표시되는 날짜 포맷 수정

rhkrwngud445 committed:
fix: 제한 글자 수 수정

rhkrwngud445 committed:
fix: 제한 글자 수 수정

rhkrwngud445 committed:
fix: 홈 lazyRow 키 값 변경

rhkrwngud445 committed:
fix: 홈 lazyRow 키 값 변경

stopstone commented on Near/app/src/main/java/com/alarmy/near/presentation/feature/friendprofileedittor/uistate/FriendProfileEditorUIState.kt:
```diff
@@ -52,7 +52,7 @@ fun FriendProfileEditorUIState.toModel(
         contactFrequency =
         contactFrequency,
-        birthday = birthday.value?.replace(".", "-"),
+        birthday = birthday.value?.replace(".", "."),
```
이 부분을 확인해보니 UI에서 보여줄때 변환이 이뤄지기 때문에 서버에 요청을 보낼 때 변환을 한번 더 해주는 것은 어떨까요? 혹은 Date, Timestamp값을 UI에 보여줄때만 변환하고 고유 값은 변경하지 않아도 좋을 것 같습니다!

수정에서 날짜를 변경하여 수정하면 400 Bad Request 로그를 확인할 수 있었습니다 🥲
다른 정보를 변경하여 수정요청을 하면 문제 없이 잘 수정되고 있어요!
스웨거를 확인해보니 다음과 같은 스펙을 가지고 있습니다!

```
  "birthday": "1997-11-19",
  "anniversaryList": [
    {
      "id": 4,
      "title": "test2",
      "date": "1997-11-19"
    }
```

rhkrwngud445 replied:
오 버그 캐치 감사합니다! 400에러가 나고 있었군요😂
말씀주신 방향으로 수정할게요!

stopstone commented on Near/app/src/main/java/com/alarmy/near/presentation/feature/home/HomeScreen.kt:
```diff
@@ -156,115 +145,129 @@ internal fun HomeScreen(
+                }
+            }
+
+            Spacer(modifier = Modifier.weight(1f))
```
| 이미지1 | 이미지2 |
| --- | --- |
| <img src="https://github.com/user-attachments/assets/ebb081c5-ae2b-4911-9462-f5c3be7a9932" width="300"/> | <img src="https://github.com/user-attachments/assets/0606c7c8-5f1a-403b-806f-39258819371a" width="300"/> |
| 갤럭시 플립 5 | 픽셀 7 |

홈 스크린의 바텀시트와 상단부분 고정, 중앙부를 1f로 두었을 때, 화면입니다!
아마 일반적인 기기에서는 여백이 넓어보이지 않는데 

폴더블같은 경우 일반 기기보다 길이가 긴 편이라 제가 보기엔 살짝 길어보이는 것 같아요! 이 부분은 한번 말씀드려보고 확인받는 것은 어떨까요?
제가 보기에만 조금 넓어보이는 것일수도 있습니다..! 😂

rhkrwngud445 replied:
여백의 미.. 라기엔 플립은 여백이 좀 크긴하군요..😅
저도 공유드려보는 것에 찬성입니다! 별도 대응도 염두에 두고 여쭤볼게요!
의견 감사합니다!!

stopstone reviewed (commented):
작업이 많았는데, 고생 많으셨습니다!
추가적으로 코드에서 설명드리에 모호한 부분 설명 드리겠습니다

| 이미지1 | 이미지2 | 이미지3 | 이미지4 |
| --- | --- | --- | --- |
| <img src="https://github.com/user-attachments/assets/a92674f0-9ef0-46d0-9006-6911a69a3e48" width="200"> | <img src="https://github.com/user-attachments/assets/bdd60f00-b099-44cd-a369-dbe7109ad178" width="200"> | <img src="https://github.com/user-attachments/assets/bdcd884f-dfce-4206-aaa6-9da4a3ccc5e4" width="200"> | <img src="https://github.com/user-attachments/assets/3f915943-f8f1-4116-9beb-81c7d55f144a" width="200"> |
| CASE 1 | CASE 2 | CASE 3 | CASE 4 |

CASE 1: 추가된 연락처의 최초(?) 상세 보기로 진입 시 {날짜} 더 가까워졌어요 형식에서 날짜가 표시되지 않는 현상이 있습니다. 재 진입했을 경우는 잘 나오고 있습니다..!

CASE 2: 연락 주기의 label과 텍스트필드 사이 간격이 다른 부분과 차이가 있습니다! 전에 이슈에 올려둔다는 것을 잊고 있었네요 😂

CASE 3: 탭 클릭 시 리플 효과 색 수정 이 부분 역시 이슈 올려둔다는 것을... 잊었습니다 두번 작업하게 해드렸네요 😭

CASE 4: RecordItem 모양이 일정하지 않는 현상이 있습니다! 애뮬레이터와 프리뷰에서는 잘 나오는데 제가 사용하고 있는 기기에서 발견했어요! 아무래도 폴더플폰이.. 크기가 일정하지 않아 고려할 부분이 많은 것 같네요 🤯

rhkrwngud445 committed:
fix: 친구 정보 수정시 생일 포맷 변경

rhkrwngud445 committed:
fix: 친구 정보 수정시 생일 포맷 변경

rhkrwngud445 committed:
fix: 친구 삭제시 이번달 챙길 사람도 제외되도록 수정

rhkrwngud445 committed:
fix: 친구 삭제시 이번달 챙길 사람도 제외되도록 수정

rhkrwngud445 committed:
feat: 그리팅 텍스트 정렬 위치 수정

rhkrwngud445 committed:
feat: 그리팅 텍스트 정렬 위치 수정

stopstone committed:
feat: 공용 바텀시트 컴포넌트 추가

- 재사용 가능한 `NearBottomSheet` 컴포posable을 구현했습니다.

stopstone committed:
refactor: `CycleSettingBottomSheet`을 공통 컴포넌트로 리팩토링

stopstone committed:
refactor: 주기 설정 BottomSheet 공용 컴포넌트로 분리 및 개선

- 기존 `FriendProfileEditor` 화면에서만 사용하던 `ReminderIntervalBottomSheet`를 삭제했습니다.
- 여러 화면에서 재사용 가능하도록 `CycleSettingBottomSheet`라는 이름의 공용 컴포넌트를 새로 만들었습니다.
- 새로운 BottomSheet는 상태 관리를 개선하고, 선택된 주기에 따라 미리보기 UI가 동적으로 변경되도록 수정했습니다.
- `FriendProfileEditorScreen`에서 기존 BottomSheet를 새로운 `CycleSettingBottomSheet`로 교체 적용했습니다.

stopstone committed:
fix: 다음 연락 주기 텍스트의 공백 추가

- `CycleSettingBottomSheet` 컴포넌트에서 "다음 연락 주기:" 접두사와 날짜 사이에 공백을 추가하여 가독성을 개선했습니다.

stopstone committed:
feat: `NearBottomSheet` 미리보기 추가

`@Preview` 어노테이션을 사용하여 `NearBottomSheet` 컴포저블의 미리보기를 추가했습니다.

stopstone committed:
chore: 바텀시트 핸들러 코너 값 수정

- NearBottomSheet 핸들러의 모서리 둥글기 값을 100.dp에서 2.5.dp로 수정했습니다.

stopstone committed:
refactor: 주기 설정 BottomSheet 공용 컴포넌트로 분리

- `friendcontactcycle`과 `friendprofileedittor` 피처에서 각각 사용하던 `CycleSettingBottomSheet`를 `ui/component/bottomsheet` 패키지로 이동하여 공용 컴포넌트로 만들었습니다.
- 중복 코드를 제거하고 재사용성을 높였습니다.

stopstone committed:
refactor: NearBottomSheet의 content 타입을 ColumnScope로 변경

rhkrwngud445 committed:
feat: monthlyFriend키 주석처리

rhkrwngud445 committed:
feat: monthlyFriend키 주석처리

rhkrwngud445 committed:
fix: 챙김 기록이 적용안되는 문제 수정

rhkrwngud445 committed:
fix: 챙김 기록이 적용안되는 문제 수정

rhkrwngud445 committed:
fix: 리플 색상 변경

rhkrwngud445 committed:
fix: 리플 색상 변경

rhkrwngud445 committed:
Merge pull request #59 from near-Contact-Reminder/feat/#56-monthly-reminder

[FEAT] 이번달 챙길사람 기능

rhkrwngud445 committed:
Merge pull request #61 from near-Contact-Reminder/qa/#57-feature-contact-cycle

[QA] 연락처 주기 설정 바텀시트 날짜 형식 수정

rhkrwngud445 committed:
Merge remote-tracking branch 'origin/qa/#58-qa_except_contact' into qa/#58-qa_except_contact

# Conflicts:
#	Near/app/src/main/java/com/alarmy/near/presentation/feature/home/HomeScreen.kt

rhkrwngud445 committed:
chore: 버전 업
